File: src/main/java/com/example/tugascrud/repository/KtpRepository.java
package com.example.tugascrud.repository;


import com.example.tugascrud.model.entity.Ktp;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface KtpRepository extends JpaRepository<Ktp, Integer> {

    Optional<Ktp> findByNomorKtp(String nomorKtp);

}

File: src/main/java/com/example/tugascrud/repository/KtpRepository.kt
package com.example.tugascrud.repository

import com.example.tugascrud.model.entity.KtpEntity
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface KtpRepository : JpaRepository<KtpEntity, Int> {

    fun existsByNomorKtp(nomorKtp: String): Boolean

    fun findByNomorKtp(nomorKtp: String): Optional<KtpEntity>
}